Transformer/LuceneToBddTransformer.php

```diff
@@ -0,0 +1,183 @@
+<?php
+
+namespace OpenOrchestra\Transformer;
+
+use OpenOrchestra\Transformer\LuceneToBddTransformerInterface;
+use OpenOrchestra\Exception\MissingFieldTransformerException;
+
+/**
+ * Class LuceneToBddTransformer
+ */
+class LuceneToBddTransformer implements LuceneToBddTransformerInterface
+{
+    protected $field = null;
+
+    /**
+     * @param string $field
+     */
+    public function setField($field)
+    {
+        $this->field = $field;
+    }
+
+    /**
+     * @param string $value
+     *
+     * @throws MissingFieldTransformerException
+     *
+     * @return array
+     */
+    public function transform($value)
+    {
+        if (is_array($value) && is_string($this->field) && array_key_exists($this->field, $value)) {
+            $value[$this->field] = $this->transformField(json_decode($value[$this->field], true));
+
+            return $value;
+        }
+
+        throw new MissingFieldTransformerException();
+    }
+
+    /**
+     * @param array $value
+     *
+     * @throws MissingFieldTransformerException
+     *
+     * @return array
+     */
+    public function reverseTransform($value)
+    {
+        if (is_array($value) && is_string($this->field) && array_key_exists($this->field, $value)) {
+            $value[$this->field] = preg_replace('/ (\+|-)/', '$1', $value[$this->field]);
+            $value[$this->field] = $this->reverseTransformField($value[$this->field]);
+
+            return $value;
+        }
+
+        throw new MissingFieldTransformerException();
+    }
+
+    /**
+     * @param string $condition
+     *
+     * @return array|string
+     */
+    public function transformField($conditions)
+    {
+        if (!is_null($conditions)) {
+            foreach($conditions as $key => $condition) {
+                if(is_array($condition)){
+                    $conditions[$key] = $this->transformField($condition);
+                }
+                if ($key === '$eq' || $key === $this->field) {
+                    $conditions = $conditions[$key];
+                    break;
+                } elseif ($key === '$ne') {
+                    $conditions = '-'.$conditions[$key];
+                    break;
+                } elseif ($key === '$and') {
+                    $conditions = '(+'.implode('+', $conditions[$key]).')';
+                    $conditions = preg_replace('/\+-/', '-', $conditions);
+                    break;
+                } elseif ($key === '$or') {
+                    $conditions = '('.implode(' ', $conditions[$key]).')';
+                    break;
+                }
+            }
+        }
+
+        return $conditions;
+    }
+
+    /**
+     * @param string $condition
+     *
+     * @return string
+     */
+    public function reverseTransformField($condition, $count = 0, $aliases = array(), $delimiter = '##')
```

Comment:
If this method is only used internally to the transformer, you should put it protected.